src/contracts/NFT.ts

```diff
@@ -70,22 +74,32 @@ export class Nft extends SmartContract {
     )
     if (estimateGas) return <G extends false ? string : BigNumber>estGas
 
+    const addresses = [minter, paymentCollector, mpFeeAddress, feeToken]
+    if (accessListContract) {
```

Comment:
only templateID == 4 supports accessLists , so pls make sure that we don't pass it to other template, because those templates might use those fields for something else